search/cities_boundaries_table.cpp

```diff
@@ -0,0 +1,94 @@
+#include "search/cities_boundaries_table.hpp"
+
+#include "search/categories_cache.hpp"
+#include "search/localities_source.hpp"
+#include "search/mwm_context.hpp"
+#include "search/utils.hpp"
+
+#include "indexer/cities_boundaries_serdes.hpp"
+
+#include "coding/reader.hpp"
+
+#include "base/assert.hpp"
+#include "base/cancellable.hpp"
+#include "base/logging.hpp"
+
+#include <algorithm>
+
+using namespace indexer;
+using namespace std;
+
+namespace search
+{
+// CitiesBoundariesTable::Boundaries ---------------------------------------------------------------
+bool CitiesBoundariesTable::Boundaries::HasPoint(m2::PointD const & p) const
+{
+  return any_of(m_boundaries.begin(), m_boundaries.end(),
+                [&](CityBoundary const & b) { return b.HasPoint(p, m_eps); });
+}
+
+// CitiesBoundariesTable ---------------------------------------------------------------------------
+bool CitiesBoundariesTable::Load(Index const & index)
```

Comment:
`#include "indexer/index.hpp`

Reply:
No, I don't need ```index.hpp```, because index is just forwarded to FindWorld, and not used here.

Reply:
Ok